packages/core/core/src/registries/config.ts

```diff
@@ -1,22 +1,39 @@
 import type { ConfigProvider } from '@strapi/types';
-import _, { PropertyName } from 'lodash';
+import { get, set, has, isString, type PropertyName } from 'lodash';
+import type { LoadedStrapi } from '../Strapi';
 
 type Config = Record<string, unknown>;
 
-export default (initialConfig = {}): ConfigProvider => {
+export default (initialConfig = {}, strapi?: Partial<LoadedStrapi>): ConfigProvider => {
```

Comment:
Why use `Partial<LoadedStrapi>` instead of `Strapi`? 

Also, why not use `Strapi | LoadedStrapi` to represent accurately what can be expected?